_posts/en/newsletters/2019-05-29-newsletter.md

```diff
@@ -0,0 +1,631 @@
+    transactions during fee events.
+
+    The 30-minute presentation covers each point concisely, making it an
+    excellent high-level overview for anyone interested in learning
+    about the Bitcoin fee market and how to mitigate expected fee
+    increases.
+
+## Proposed transaction output commitments
+
+The [proposed][bip-coshv] opcode `OP_CHECKOUTPUTSHASHVERIFY` allows a
+Taproot address to commit to one or more tapscripts that require the
+transaction spending them to include a certain set of outputs, a
+technique that contract protocol researchers call a *covenant*.
+The primary described benefit of this proposed opcode is allowing a
+small transaction to be confirmed now (when fees might be high) that
+trustlessly guarantees a set of people will receive their actual
+payments later when fees might be lower.  This can make it much more
+economical for organization that already implement techniques such as
+[payment batching][] to handle sudden fee spikes.
+
+Before we look at the new opcode itself, let's take a moment to look at
+how you might accomplish something similar using current Bitcoin
+transaction features.
+
+Alice wants to pay a set of ten people but transaction fees are
+currently high so that she doesn't want to send ten separate
+transactions or even use payment batching to send one transaction
+that includes an output for each of the receivers.  Instead, she wants
+to trustlessly commit to paying them in the future without having to pay
+onchain fees for ten outputs now.  So Alice asks each of the receivers
+for one of their public keys and creates an unsigned and unbroadcast *setup* transaction
+that pays those keys using a 10-of-10 multisig script.  Then she creates
+a child transaction transaction that spends from the multisig output to the
+10 outputs she originally wanted to create.  We'll call this child
+transaction the *distribution* transaction.  She asks all the receivers
+to sign this distribution transaction and she ensures each person
+receives everyone else's signatures, then she signs and broadcasts the
+setup transaction.
+
+![A setup transaction paying a pre-signed distribution transaction](/img/posts/2019-05-tx-tree-1.png)
+
+When the setup transaction receives a reasonable number of
+confirmations, there's no way for Alice to take back her payment to the
+10 receivers.  As long as each of the receivers has a copy of the
+distribution transaction and all the others' signatures, there's also no
+way for any receivers to cheat any other receiver out of a payment.  So
+even though the distribution transaction that actually pays the
+receivers hasn't been broadcast or confirmed, the payments have the same
+amount of security as the confirmed setup transaction.  At any time, any
+of the receivers who wants to spend their money can broadcast
+the distribution transaction and wait for it to confirm (perhaps helping
+to pay its fee using child-pays-for-parent fee bumping).
```

Comment:
This is a bit tricky. We don't have package relay yet, so if the distribution transaction doesn't have enough fees to get into the mempool, then there's no way to bump it. That's an extreme case - most of the time I expect the distribution transaction's fee to be somewhere between the mempool minimum and the rate required to be mined. It might be worth adding that in the case of extreme fee spikes, the distribution transaction would be stranded until mempools emptied enough to accept it.